rdrf/rdrf/admin.py

```diff
@@ -178,6 +183,87 @@ def get_readonly_fields(self, request, obj=None):
         return () if obj is None else ("code",)
 
 
+class ActivationKeyExpirationListFilter(admin.SimpleListFilter):
+    title = _('activation key expired')
+    parameter_name = 'activation_key_expired'
+
+    def lookups(self, request, model_admin):
+        return [
+            ('True', _('Yes')),
+            ('False', _('No')),
+        ]
+
+    def queryset(self, request, queryset):
+        expired_profiles = [profile.id for profile in queryset if profile.activation_key_expired()]
+
+        if self.value() == 'True':
+            return queryset.filter(id__in=expired_profiles)
+        elif self.value() == 'False':
+            return queryset.exclude(id__in=expired_profiles)
+        else:
+            return queryset
+
+
+def resend_activation_mail(profile, site, request=None):
+    if not hasattr(profile.user, 'registrationprofile') or profile.activated:
+        return False
+
+    profile.create_new_activation_key()
```

Comment:
This new activation key doesn't work for the user if their previous activation key has expired.

The original registration model logic to `activate_user` will check if the `activation_key_expired` is true, and if so will not allow the profile to be activated. So if the user you 'resend' the activation email to has joined more than `settings.ACCOUNT_ACTIVATION_DAYS` ago, they will not be able to successfully activate their account.